src/control/server/storage/bdev/backend.go

```diff
@@ -302,12 +310,58 @@ func detectVMD() ([]string, error) {
 	return vmdAddrs, nil
 }
 
-// Prepare will execute SPDK setup.sh script to rebind PCI devices as selected
-// by bdev_include and bdev_exclude white and black list filters provided in the
-// server config file. This will make the devices available though SPDK.
+func getRemoveHugePageFn(hugePageDir, prefix, tgtUsr string) filepath.WalkFunc {
+	return func(path string, info os.FileInfo, err error) error {
+		switch {
+		case err != nil:
+			return err
+		case info.IsDir():
+			if path != hugePageDir {
+				return filepath.SkipDir // skip subdirectories
+			}
+			return nil
+		case !strings.HasPrefix(info.Name(), prefix):
+			return nil // skip files without prefix
+		}
+
+		uid := info.Sys().(*syscall.Stat_t).Uid
+		uidStr := strconv.FormatUint(uint64(uid), 10)
+		usr, err := user.LookupId(uidStr)
+		if err != nil {
+			return err
+		}
+		if usr.Username != tgtUsr {
+			return nil // skip not owned by target user
+		}
```

Comment:
Thanks for the suggestions, had to make a fair amount of modifications to make this work, testing the WalkFunc callback outside of the Walk function means that you have to test results of call for each test input individually, I moved expErr into the test input and evaluated the removed list after iterating through all the test Inputs. because you can't test the result of iterating through a tree/subdirectories in this manner I just tested the setting of skip directory "special" error on a single file info input (using the previous technique it would be possible to see that matching files in a subdirectory would be skipped).